### PR TITLE
Expose tracef in zig template

### LINKED ISSUE #244: (zig) Add `tracef` API call  (APIs should have feature parity)
The C API has `tracef`, so should Zig.

### PR DESCRIPTION
**Pull request by sporksmith**
Fixes #244

### CODE REVIEW

**Comment by sporksmith** [comment]
This is the "light" solution. I'm going to send an alternate PR shortly with a safer and more idiomatic interface, but it requires a bit more code.

**Comment by sporksmith** [comment]
> This is the "light" solution. I'm going to send an alternate PR shortly with a safer and more idiomatic interface, but it requires a bit more code.

After further consideration, I'm just proposing this PR for now. I think the safe wrapper around `tracef` is too complex to belong here. (One minor snag I ran into is it doesn't appear possible to both declare an `extern` `tracef` and provide a public function named `tracef` in the same file)

**Comment by sporksmith** [comment]
As a point of comparison, this is pretty much how the zig standard library expose's libc's `printf`: https://github.com/ziglang/zig/blob/4f4f0bc6f03a84597247504a5ee15ed7f1e050d1/lib/std/c.zig#L105

**Comment by joshgoebel** [comment]
>  it doesn't appear possible to both declare an extern tracef and provide a public function named tracef in the same file)

Yeah I was worried about that.

> As a point of comparison, this is pretty much how the zig standard library expose's libc's printf:

Then I'd say we're in good company. :)

File: cli/assets/templates/zig/src/wasm4.zig [hunk 1]
```diff
@@ -121,3 +121,7 @@ pub fn trace(x: []const u8) void {
     traceUtf8(x.ptr, x.len);
 }
 extern fn traceUtf8(strPtr: [*]const u8, strLen: usize) void;
+/// Use with caution, as there's no compile-time type checking.
+/// %d and %f expects 32-bit integers and floats, respectively.
```
Comment (joshgoebel) [inline cli/assets/templates/zig/src/wasm4.zig:125]: I thought I commented, but now I don't see it.  Are you certain?  Doesn't it take `f64` not `f32`??? We move the pointer 8 bytes for floats in the runtime code.

Comment (sporksmith) [reply cli/assets/templates/zig/src/wasm4.zig:125]: Oops, you're right!

Comment (sporksmith) [reply cli/assets/templates/zig/src/wasm4.zig]: Fixed

**Review by joshgoebel — APPROVED** [review]
Much better!

**Comment by aduros** [comment]
Thanks!

**Pull request merged**